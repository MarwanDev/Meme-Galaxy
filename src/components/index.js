export { default as Navbar } from './Navbar/Navbar';
export { default as Header } from './Header/Header';
export { default as Categories } from './Categories/Categories';
export { default as Layout } from './Layout/Layout';
export { default as Category } from './Category/Category';
