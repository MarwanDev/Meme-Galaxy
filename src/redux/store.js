import { configureStore } from "@reduxjs/toolkit";
import memesReducer from "./memesSlice/memesSlice";

const store = configureStore({
  reducer: {
    meme: memesReducer,
  },
});

export default store;
