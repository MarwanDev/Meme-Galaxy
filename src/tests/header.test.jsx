import { Header } from "../components";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import renderer from "react-test-renderer";
import store from "../redux/store";
import { Provider } from "react-redux";

describe("Header", () => {
  it("renders header correctly", () => {
    const header = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });
  it("renders header correctly", () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    );
  });
  it("displays the Loading text", () => {
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    const loadingElement = screen.getByText("Loading...");
    expect(loadingElement).toBeInTheDocument();
  });
});
