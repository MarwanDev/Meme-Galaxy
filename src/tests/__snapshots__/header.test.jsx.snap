// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders header correctly 1`] = `
<div
  className="header-container"
>
  <h2
    style={
      {
        "backgroundColor": "#4369b2",
        "color": "#fff",
        "marginBottom": 15,
        "textAlign": "center",
      }
    }
  >
    memes stat
  </h2>
  <ul
    className="info-container"
  >
    <li
      className="info"
    >
      <h4>
        0
      </h4>
      <h3>
        memes
      </h3>
    </li>
    <li
      className="info"
    >
      <h4>
        -Infinity
      </h4>
      <h3>
        categories
      </h3>
    </li>
    <li
      className="info"
    >
      <h4>
        -Infinity
      </h4>
      <h3>
        authors
      </h3>
    </li>
  </ul>
</div>
`;
